packages/rollup-contracts/contracts/FullStateManager.sol

```diff
@@ -123,10 +144,44 @@ contract FullStateManager is StateManager {
      * @param _codeContractAddress The address of the code contract.
      * @return The hash of the bytecode at this address.
      */
-    function getCodeContractHash(address _codeContractAddress) internal view returns (bytes32 _codeContractHash) {
+    function getCodeContractHash(address _codeContractAddress) public view returns (bytes32 _codeContractHash) {
         // TODO: Look up cached hash values eventually to avoid having to load all of this bytecode
         bytes memory codeContractBytecode = getCodeContractBytecode(_codeContractAddress);
         _codeContractHash = keccak256(codeContractBytecode);
         return _codeContractHash;
     }
+
+    /**
+     * @notice Deploys a code contract, and then registers it to the state
+     * @param _ovmContractInitcode The bytecode of the contract to be deployed
+     * @return the codeContractAddress.
+     */
+    function deployContract(
+        address _newOvmContractAddress,
+        bytes memory _ovmContractInitcode
+    ) public returns(address codeContractAddress) {
+        // Safety check the initcode, unless the overrideSafetyChecker flag is set to true
+        if (!overrideSafetyChecker && !safetyChecker.isBytecodeSafe(_ovmContractInitcode)) {
```

Comment:
not within the scope of this PR, but at some point, think we'll need to pull out the safety checking into a separate step since the gas cost of safety checking is in the 5-10m range for large contracts

Reply:
Yeah 100%. This'll have to be added into the partial state manager

Reply:
Am looking to deprecate this full state manager once we get to a point where we can deprecate the `rollup-full-node` cuz we won't need it anymore for L2 once we use Geth